fdbmonitor/fdbmonitor_lib.cpp

```diff
@@ -18,7 +18,9 @@
  * limitations under the License.
  */
 
+#include <csignal>
```

Comment:
1. This is not guarded by `#if defined(__APPLE__) || defined(__FreeBSD__)` unlike the change in `fdbmonitor.cpp`. Why is that?
2. In general, can we move this to `fdbmonitor.h`? Looks like it's needed both in `fdbmonitor.cpp` and `fdbmonitor_lib.cpp`, both of which include `fdbmonitor.h`. We can also keep it this way if there's nothing in `fdbmonitor.h` that needs `<csignal>`.

Reply:
For (1), not sure about the guard issue since `csignal` and `sys.time.h` are both part of C/C++ standard library, and should be implemented by all compilers over all platforms (even I remembered that Mac does something strange over some of the std libraries,) For (2), I do not have any opinion.

Reply:
Let's make them consistent i.e. either guard or not guard across both files.

Reply:
Hi there, thanks for all the comments ^-^

For 2, I think it wouldn't matter much since there are only two files, but I followed the general advice of not including headers that won't be needed everywhere. `fdbmonitor_lib.cpp` uses `gettimeofday` and `fdbmonitor.cpp` uses kqueue related functions. Not sure what's the general convention in fdb but if `fdbmonitor.h` includes all headers that both needs, compiler/preprocessor would do unnecessary work to include `sys/time.h` and `sys/event.h` into files that these won't be not needed. Probably a nitpick for this change though, but generally it might be better to keep includes in the places they are used.

Reply:
> Let's make them consistent i.e. either guard or not guard across both files.

Oh I think `csignal` can come out of the guard, but it makes sense to me to keep `sys/event.h` in the guard, because that's only needed on those systems (due to kqueue). `gettimeofday` and `signal` & `sigprocmask` are used for all paths, so they should be outside (I think it might be compiling on Linux without `csignal` being outside due to it might be included transitively).

Reply:
>> I followed the general advice of not including headers that won't be needed everywhere

Agreed, if `fdbmonitor.h` does not need `csignal`, we can include it separately in `fdbmonitor.cpp` and `fdbmonitor_lib.cpp`.

>> Oh I think csignal can come out of the guard

Sounds good.